packages/fxa-content-server/tests/functional/settings/change_email.js

```diff
@@ -52,8 +51,8 @@ registerSuite('settings change email', {
         .then(fillOutEmailFirstSignUp(email, PASSWORD))
         .then(testElementExists(selectors.CONFIRM_SIGNUP_CODE.HEADER))
         .then(fillOutSignUpCode(email, 0))
-        .then(waitForUrl((url) => !url.includes('settings')))
```

Comment:
I added this because the following `testElementExists` was (sometimes) firing before the page navigated to `/settings` but apparently it wasn't enough. IDK that we need it but I'm not optimistic your change will help.

Reply:
Hrm let me try the `visibleByQSA` selector, maybe that would help.